lib/Mojolicious/Static.pm

```diff
@@ -33,6 +34,9 @@ sub dispatch {
   $path = $stash->{path} ? $path->new($stash->{path}) : $path->clone;
   return undef unless my @parts = @{$path->canonicalize->parts};
 
+  # Don't serve when a part has embedded path separators
+  splitdir(canonpath($_)) > 1 && return undef for @parts;
```

Comment:
Might be better to do @{[splitdir canonpath $_]} since it's not documented that splitdir will return the number of components in scalar context.